trieste/models/interfaces.py

```diff
@@ -0,0 +1,229 @@
+        dimensions of ``query_points`` are all interpreted as broadcasting dimensions instead of
+        batch dimensions, and the covariance is squeezed to remove redundant nesting.
+
+        :param query_points: The points at which to make predictions, of shape [..., D].
+        :return: The mean and variance of the independent marginal distributions at each point in
+            ``query_points``. For a predictive distribution with event shape E, the mean and
+            variance will both have shape [...] + E.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def predict_joint(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        """
+        :param query_points: The points at which to make predictions, of shape [..., B, D].
+        :return: The mean and covariance of the joint marginal distribution at each batch of points
+            in ``query_points``. For a predictive distribution with event shape E, the mean will
+            have shape [..., B] + E, and the covariance shape [...] + E + [B, B].
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def sample(self, query_points: TensorType, num_samples: int) -> TensorType:
+        """
+        Return ``num_samples`` samples from the independent marginal distributions at
+        ``query_points``.
+
+        :param query_points: The points at which to sample, with shape [..., N, D].
+        :param num_samples: The number of samples at each point.
+        :return: The samples. For a predictive distribution with event shape E, this has shape
+            [..., S, N] + E, where S is the number of samples.
+        """
+        raise NotImplementedError
+
+    def predict_y(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        """
+        Return the mean and variance of the independent marginal distributions at each point in
+        ``query_points`` for the observations, including noise contributions.
+
+        Note that this is not supported by all models.
+
+        :param query_points: The points at which to make predictions, of shape [..., D].
+        :return: The mean and variance of the independent marginal distributions at each point in
+            ``query_points``. For a predictive distribution with event shape E, the mean and
+            variance will both have shape [...] + E.
+        """
+        raise NotImplementedError(
+            f"Model {self!r} does not support predicting observations, just the latent function"
+        )
+
+    def get_observation_noise(self) -> TensorType:
+        """
+        Return the variance of observation noise.
+
+        Note that this is only supported for models with with homoscedastic noise, and so this
+        function returns a scalar.
+
+        :return: The observation noise.
+        """
+        raise NotImplementedError(f"Model {self!r} does not provide scalar observation noise")
+
+    def get_kernel(self) -> gpflow.kernels.Kernel:
```

Comment:
this is a recent addition? this is another model-specific dependency  that we should probably remove, that is, the output type, not the method - any suggestions?

Reply:
I could just say its a Callable? But I don't envisage us every using a model that has  a kernel that isnt a gpflow one (e.g. even gpflux uses the gpflow kernels)

Reply:
right, that would be a generic thing to do here, though its not that useful any more as a type check

does this really need to be a part of this class? it assumes a nonparametric probabilistic model, no? perhaps it could be a method in ` GPflowPredict` which is sort of an interface for gpflow models? - there it would be fully appropriate

Reply:
You are right that get_kernel should be in the Gpflow class. I will move that and get_observation noise into  Gpflow class

Reply:
you may find that acquisition functions don't type check if you do that. Having looked at where `get_kernel` is used, I'm not sure what you can do about that other than raise a `TypeError` when you try to use random fourier.

Reply:
but don't let me discourage you

Reply:
@henrymoss I'll wait for the merge of your PR to merge this one then

Reply:
@henrymoss while you're doing that, `get_kernel` _returns_ the error where it should raise it